### PR TITLE
Build order sections

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- make sections used for build order a parameter
- update makefile to use the environment for pre-commit

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -88,19 +88,19 @@ environment:    ## handles environment creation
 dev: clean		## install the package's development version to a fresh environment
 	conda env create -f environment.yaml --name $(CONDA_ENV_NAME) --force
 	conda run --name $(CONDA_ENV_NAME) pip install -e .
-	$(CONDA_ACTIVATE) percy && pre-commit install
+	conda run --name $(CONDA_ENV_NAME) pre-commit install
```
Comment (schuylermartin45) [inline Makefile:91]: Today I learned. I assume you tested this by blowing away an re-creating an environment?

If that's the case should we delete `CONDA_ACTIVATE` then?

File: Makefile [hunk 2]
```diff
@@ -88,19 +88,19 @@ environment:    ## handles environment creation
...
 
 pre-commit:     ## runs pre-commit against files. NOTE: older files are disabled in the pre-commit config.
 	pre-commit run --all-files
 
 test:			## runs test cases
-	$(PYTHON3) -m pytest --capture=no percy/tests/
+	conda run --name $(CONDA_ENV_NAME) python -m pytest --capture=no percy/tests/
```
Comment (schuylermartin45) [inline Makefile:97]: Should we delete `PYTHON3` or perhaps roll `conda run --name $(CONDA_ENV_NAME) python` into it's own convenience variable?

I think `help` still uses `$(PYTHON3)`

File: environment.yaml [hunk 3]
```diff
@@ -18,3 +18,5 @@ dependencies:
   - conda-build
   - jsonschema
   - pre-commit
+  - pytest
+  - pytest-cov
```
Comment (schuylermartin45) [inline environment.yaml:22]: These are both already included, on lines 8 and 9

File: Makefile (hunk 2 above)
Comment (schuylermartin45) [reply Makefile:97]: I think the GitHub actions failure is something I ran into previously. I think I have a commit message somewhere that explains why I did it this way.

Comment (schuylermartin45) [reply Makefile:97]: Here's the commit for historical context: https://github.com/anaconda-distribution/percy/commit/3897afdbc4d37d8ff4da77ba35d118f4ad344fe9#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52

The comment above `PYTHON3` describes what is trying to be done/what I ran into:
```
# Resolve the path to `python3` and store it in a variable. Always invoke
# `python` using this variable. This prevents a nasty scenario where the GitHub
# Actions container fails to find `python` or `python3`, when running `make`
# commands.
```

Comment (cbouss) [reply Makefile:85]: Well my developer experience was that `make test` was not runnable, effectively preventing me from committing.
I'll try to find something that work.

Comment (schuylermartin45) [reply Makefile:85]: Did you re-make your developer environment with `make dev`?

Comment (schuylermartin45) [reply Makefile:85]: My best current guess is your local `percy` environment was stale/never updated after I had made these changes.

Comment (schuylermartin45) [reply Makefile:85]: If you did do that, I wonder if your `percy` environment is resolving to another `python3` installed on the system and not the one under `miniconda`, which would have all the test dependencies installed.

Here's my path, when under the `percy` environment
```
/Users/smartin/anaconda_3/anaconda3/envs/percy/bin/python3
```

Comment (cbouss) [reply Makefile:85]: I did run make dev. I can run make test as well with no issue. 
pre-commit however finds the python from homebrew.

Comment (schuylermartin45) [reply Makefile:85]: I sent you some other ideas over slack but I ran into this before.

I was able to get the Percy environment to point to the right Python version by opening a new terminal and running `conda activate percy`.

I've run into this a few times in other conda environments and it seems to be a semi known issue when I talked to the conda team a few months back.

**Comment by cbouss** [comment]
Will redo another PR without the changes to the makefile.

**Pull request closed without merging**